infra-cdk/lib/stacks/IAMStack.ts

```diff
@@ -3,11 +3,14 @@ import * as iam from 'aws-cdk-lib/aws-iam';
 import { Construct } from 'constructs';
 import { ENVIRONMENT } from '../core/constants';
 import { BaseStackProps } from '../core/types';
+import { S3BucketStack } from '../stacks/ReportsBucketStack';
 
 export type IAMStackProps = {
     /** DynamoDB table name to provide permissions to */
-    tableName: string;
-    reportsTableName: string;
+    tableName?: string;
+    reportsTableName?: string;
+    /** S3 bucket name to provide permissions to */
+    bucketStack?: S3BucketStack;
```

Comment:
Instead of accepting a `bucketStack: S3BucketStack`, could this accept a `reportsBucketName: string`? this provides a more descriptive variable name, and makes it flexible if we use different stacks in the future